server/sampleData/utils/helpers.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+//@ts-ignore
+import moment from 'moment';
+import readline from 'readline';
+import { Request } from 'hapi';
+
+import fs from 'fs';
+import { createUnzip } from 'zlib';
+//@ts-ignore
+import { CallClusterWithRequest } from 'src/legacy/core_plugins/elasticsearch';
+
+const BULK_INSERT_SIZE = 500;
+
+export const loadSampleData = (
+  filePath: string,
+  indexName: string,
+  req: Request,
+  callWithRequest: CallClusterWithRequest
+) => {
+  return new Promise((resolve, reject) => {
+    let count: number = 0;
+    let docs: any[] = [];
+    let isPaused: boolean = false;
+    let offset = 0;
+    const startTime = moment(new Date().getTime()).subtract({ days: 7 });
+
+    const readStream = fs.createReadStream(filePath);
+    const lineStream = readline.createInterface({
+      input: readStream.pipe(createUnzip()),
+    });
+
+    const onClose = async () => {
+      if (docs.length > 0) {
+        try {
+          await bulkInsert(docs);
+        } catch (err) {
+          reject(err);
+          return;
+        }
+      }
+      resolve(count);
+    };
+    lineStream.on('close', onClose);
+    lineStream.on('pause', async () => {
+      isPaused = true;
+    });
+    lineStream.on('resume', async () => {
+      isPaused = false;
+    });
+    lineStream.on('line', async (doc) => {
+      // for the initial doc, get the timestamp to properly set an offset
+      if (count === 0) {
+        const initialTime = moment(JSON.parse(doc).timestamp);
+        offset = startTime.diff(initialTime);
+      }
+      count++;
+      docs.push(doc);
+
+      if (docs.length >= BULK_INSERT_SIZE && !isPaused) {
+        lineStream.pause();
+
+        // save the docs to insert, and empty out current docs list
+        const docsToInsert = docs.slice();
+        docs = [];
+        try {
+          await bulkInsert(docsToInsert);
+          lineStream.resume();
+        } catch (err) {
+          lineStream.removeListener('close', onClose);
+          lineStream.close();
+          reject(err);
+        }
+      }
+    });
+
+    const bulkInsert = async (docs: any[]) => {
+      try {
+        const bulkBody = prepareBody(docs, offset);
+        const resp = await callWithRequest(req, 'bulk', {
+          body: bulkBody,
+        });
+        if (resp.errors) {
```

Comment:
log `resp.errors` and `err` before returing

Reply:
sure, will add